src/qtviewer/ParseIfcFile.cpp

```diff
@@ -21,22 +31,145 @@ void ParseIfcFile::Parse(const std::string& filePath)
 
     IfcGeom::Iterator* it = new IfcGeom::Iterator(settings, &file);
     if (!it->initialize()) {
-        outputMsg("Error: Iterator failed to initialize! Aborting.");
+        MessageLogger::log("Error: Iterator failed to initialize! Aborting.");
         delete it;
-        return;
+        return false;
     }
 
     do {
         //const IfcGeom::BRepElement* bRepElem = it->get_native();
         const IfcGeom::TriangulationElement* triElem = static_cast<const IfcGeom::TriangulationElement*>(it->get());
-        outputMsg(triElem->type() + ": " + triElem->name());
+        MessageLogger::log(triElem->type() + ": " + triElem->name());
 
         const boost::shared_ptr<IfcGeom::Representation::Triangulation>& triElemGeom = triElem->geometry_pointer();
 
+        const std::vector<int>& elemFaces = triElemGeom->faces();
```

Comment:
`triElemGeom->material_ids()` will give you a mapping of triangle (a 3-length subarray of faces) to an index in `triElemGeom->materials()`. So if you want to split by material. You can do something like:

```C++
std::vector<std::vector<int>> grouped_faces;
auto fit = triElemGeom->faces().begin();
for (auto& mid : triElemGeom->material_ids()) {
    while (grouped_faces.size() <= mid) {
        grouped_faces.emplace_back();
    }
    for (int i = 0; i < 3; ++i) {
        grouped_faces[mid].push_back(fit++);
    }
}
```

Reply:
Thanks. Updated by mapping faces to material IDs, so now an element can display multiple materials applied to parts (like a layered composite wall).